backend/main.py

```diff
@@ -142,15 +146,15 @@ def llm_thread(g, prompt):
                 verbose=True,
                 streaming=True,
                 callback_manager=CallbackManager([ChainStreamHandler(g)]),
-                temperature=0.7,
+                temperature=os.environ['TEMPERATURE'],
                 openai_api_key=os.environ['OPENAI_API_KEY'],
                 openai_organization=os.environ['OPENAI_ORG_ID']
             )
 
             encoding = tiktoken.get_encoding(encoding_name)
 
-            # the system message gets 10 new docs. Only include 2 more for new queries
-            if len(chat) > 2:
+            # the system message gets NUM_RELEVANT_DOCS new docs. Only include NUM_RELEVANT_FOLLOWUP_DOCS more for new queries
+            if len(chat) > int(os.environ['NUM_RELEVANT_FOLLOWUP_DOCS']):
```

Comment:
The previous line `len(chat) > 2` was designed to determine whether or not there had been any previous conversation, so I think this change is incorrect (should stick with the hardcoded `2`).